### ISSUE TITLE
store fails to resolve against context during SpecEx

### ISSUE THREAD

**Issue by sjmiles** [PolymerLabs/arcs#2663]
See using the DeviceClientApi as described [here](https://docs.google.com/document/d/1j4uCFbXsZHJQuatnbCvUF_D24_KCn13FULCO9OSgxWY/edit#).

To reproduce locally, you should use this branch: https://github.com/sjmiles/arcs/tree/pipes-feb

Note: user `testuser` has existing data for `PROFILE_pipeEntities`. If you use another user, observe some entities first via `ShellApi.observeEntity`, e.g. `ShellApi.observeEntity('{"type": "address", "name": "Mountain View"}')`. 

Invoking `ShellApi.receiveEntity(`{"type": "maps"}`)` causes the recipe below to be selected by planner:
```
particle MapQuery in './source/MapQuery.js'
  in PipeEntity pipe
  in [PipeEntity] entities
  out SuggestionText text
  consume pipes
    provide pipeContent

recipe MapQuery
  use #pipe_maps as pipe
  map 'PROFILE_pipeEntities' as entities
  create #piped #text as text
  MapQuery
    entities = entities
    pipe = pipe
    text = text
```

And this error occurs during Speculative Execution:

![image](https://user-images.githubusercontent.com/98045/52151985-b2fdc180-2629-11e9-91fd-4ade75410ac3.png)

I'm not well versed in the Storage implementation, but it looks to me like `entities` becomes mapped to the Arc's `LivePipeEntity` (Variable) store instead of the Context's `PROFILE_pipeEntities`.

**Comment by piotrswigon** [comment]
@sjmiles Is it expected that I don't see 'PROFILE_pipeEntities' store when I'm loading the web-shell for head?

<img width="1479" alt="screen shot 2019-02-02 at 10 03 04 am" src="https://user-images.githubusercontent.com/26159485/52154399-c5cbd100-26d1-11e9-8a5d-ebf32e2b4b8f.png">

**Comment by shans** [comment]
I'm not getting the error during speculative execution, but I am getting 0 suggestions produced.

**Comment by shans** [comment]
Never mind, got it.

**Comment by shans** [comment]
This appears to be happening because both the PROFILE_pipeEntities and LivePipeEntity stores have the identical storageKey .. which shouldn't be possible. I'm trying to track down why that's happening now.

**Comment by shans** [comment]
Looks like we sufficiently understand why this is happening to fix it - @sjmiles has a fix in his current demo codebase.

Basically, multiple different manifests were being given the same ID/fileName by the shell on parse, which was eventually causing a collision.

**Comment by sjmiles** [comment]
Fixed in #2673
